docs/src/api.md

```diff
@@ -81,6 +81,18 @@ a = random_agent(model)
 sort!(collect(nearby_agents(a, model)))
 ```
 
+## Higher-order interactions
+
+There may be times when pair-wise, triplet-wise or higher interactions need to be
+accounted for across most or all of the model's agent population. The following methods
+provide an interface for such calculation.
+
+```@docs
+iter_agent_groups
+map_agent_groups
+index_groups_filtered
```

Comment:
I don't think the `_filtered` part at the end should be here. I propose to rename this function to `index_mapped_groups`. I would be using this function even without hte `filter` argument, to get a convenience vector that maps results of `map_agent_groups` to agents used in the mapping.